tests/antares_historic/test_antares_historic.py

```diff
@@ -50,20 +132,43 @@ def input_library(
         return parse_yaml_library(lib)
 
 
-@pytest.mark.skip("Missing max operator in modeleur to read thermal model")
 def test_basic_balance_using_yaml(
-    study_component: InputStudy, input_library: InputLibrary
+    study_component_basic: InputStudy,
+    input_library: InputLibrary,
 ) -> None:
+    study_path = study_component_basic[1]
+    study_component = study_component_basic[0]
+
     result_lib = resolve_library([input_library])
 
     components_input = resolve_components_and_cnx(study_component, result_lib)
     consistency_check(components_input.components, result_lib.models)
+    database = build_data_base(study_component, study_path)
+    network = build_network(components_input)
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0, 1]), scenarios)
+    status = problem.solver.Solve()
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 150
+
+
+def test_thermal_balance_using_yaml(
+    study_component_thermal: InputStudy,
+    input_library: InputLibrary,
+) -> None:
+    study_path = study_component_thermal[1]
+    study_component = study_component_thermal[0]
 
-    database = build_data_base(study_component, None)
+    result_lib = resolve_library([input_library])
+
+    components_input = resolve_components_and_cnx(study_component, result_lib)
+    consistency_check(components_input.components, result_lib.models)
+    database = build_data_base(study_component, study_path)
     network = build_network(components_input)
 
     scenarios = 1
-    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    problem = build_problem(network, database, TimeBlock(1, [0, 1]), scenarios)
     status = problem.solver.Solve()
     assert status == problem.solver.OPTIMAL
-    assert problem.solver.Objective().Value() == 3000
+    assert problem.solver.Objective().Value() == 330
```

Comment:
If I am not mistaken, you have a thermal cost of 1.1 and unsupplied energy cost of 1, so you will never use your thermal cluster as it is cheaper not to supply energy, you should increase unsupplied energy cost. Moreover I do not understand how you get this optimal value. In the case with large unsupplied energy cost, you will produce at a cost of 1.1/MW, so total cost will be 100 * 1.1 + 50 * 1.1 = 165 ?